### PR TITLE
LR Schedulers

### PR DESCRIPTION
**Pull request by mattismegevand**
Added three learning rate scheduler:
- MultiStepLR
- ReduceLROnPlateau
- CosineAnnealingLR

### CODE REVIEW

**Comment by geohot** [comment]
Put test in test/extra, and can you add a couple integration tests that train models with these schedulers?

**Comment by mattismegevand** [comment]
Done

**Comment by geohot** [comment]
Tests failing

**Comment by mattismegevand** [comment]
Forgot to set the seed for tensors, should be good now

**Comment by geohot** [comment]
Umm, why did it fail with SGD?

**Comment by mattismegevand** [comment]
There’s no reason in particular both optimizers works with the schedulers. Since for the testing we're training on MNIST it was easy for the model to reach convergence without a scheduler. So I tried multiple settings and Adam with those parameters worked more reliably and could pass the 3 categories of test (CPU / LLVM / Torch).

Less shaky testing could be done by training on ImageNet for example but I preferred to do something lightweight that didn't require lots of training time.

**Comment by geohot** [comment]
Cool, merged!
